packages/game-passes/test/GamePasses.test.ts

```diff
@@ -0,0 +1,1251 @@
+
+      await sandboxPasses
+        .connect(admin)
+        .setDefaultRoyalty(user2.address, newRoyaltyPercentage);
+
+      const salePrice = ethers.parseEther('1');
+      const royaltyInfo = await sandboxPasses.royaltyInfo(
+        TOKEN_ID_1,
+        salePrice,
+      );
+
+      expect(royaltyInfo[0]).to.equal(user2.address);
+      expect(royaltyInfo[1]).to.equal(
+        (salePrice * newRoyaltyPercentage) / 10000n,
+      );
+    });
+
+    it('should allow admin to set token-specific royalty', async function () {
+      const tokenRoyaltyPercentage = 800n; // 8%
+
+      await sandboxPasses
+        .connect(admin)
+        .setTokenRoyalty(TOKEN_ID_2, user2.address, tokenRoyaltyPercentage);
+
+      const salePrice = ethers.parseEther('1');
+      const royaltyInfo = await sandboxPasses.royaltyInfo(
+        TOKEN_ID_2,
+        salePrice,
+      );
+
+      expect(royaltyInfo[0]).to.equal(user2.address);
+      expect(royaltyInfo[1]).to.equal(
+        (salePrice * tokenRoyaltyPercentage) / 10000n,
+      );
+    });
+  });
+
+  describe('Pause Functionality', function () {
+    it('should allow admin to pause and unpause the contract', async function () {
+      expect(await sandboxPasses.paused()).to.be.false;
+
+      await sandboxPasses.connect(admin).pause();
+      expect(await sandboxPasses.paused()).to.be.true;
+
+      await sandboxPasses.connect(admin).unpause();
+      expect(await sandboxPasses.paused()).to.be.false;
+    });
+
+    it('should not allow minting when paused', async function () {
+      await sandboxPasses.connect(admin).pause();
+
+      const price = ethers.parseEther('0.1');
+      const deadline = (await time.latest()) + 3600;
+      const nonce = 0;
+
+      // Create signature
+      const signature = await createMintSignature(
+        signer,
+        user1.address,
+        TOKEN_ID_1,
+        MINT_AMOUNT,
+        price,
+        deadline,
+        nonce,
+      );
+
+      // Try to mint while paused
+      await expect(
+        sandboxPasses
+          .connect(user1)
+          .mint(
+            user1.address,
+            TOKEN_ID_1,
+            MINT_AMOUNT,
+            price,
+            deadline,
+            signature,
+          ),
+      ).to.be.revertedWithCustomError(sandboxPasses, 'EnforcedPause');
+    });
+
+    it('should not allow transfers when paused', async function () {
+      // Mint tokens first
+      await sandboxPasses
+        .connect(admin)
+        .adminMint(user1.address, TOKEN_ID_1, MINT_AMOUNT);
+
+      // Pause the contract
+      await sandboxPasses.connect(admin).pause();
+
+      // Try to transfer while paused
+      await expect(
+        sandboxPasses
+          .connect(user1)
+          .safeTransferFrom(user1.address, user2.address, TOKEN_ID_1, 2, '0x'),
+      ).to.be.revertedWithCustomError(sandboxPasses, 'EnforcedPause');
+    });
+  });
+
+  describe('URI', function () {
+    it('should return correct token URI', async function () {
```

Comment:
does token need to exist for URI to be returned?

Reply:
It doesnt, our uri function only stores the base URI, not a per token URI.
For every token id we will return the same string with an `{id}` to be replaced on the client side